src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs

```diff
@@ -375,49 +302,101 @@ private void RefreshExpandedDisplay(Action refreshDisplay)
-                    newViewModel = nodeViewModel.DynamoViewModel.WatchHandler.GenerateWatchViewModelForData(
-                        mirrorData, null, nodeViewModel.NodeModel.AstIdentifierForPreview.Name, false);
-                },
-                (m) =>
+            if (largeContentGrid.Children.Count == 0)
+            {
+                var tree = new WatchTree
                 {
-                    if (largeContentGrid.Children.Count == 0)
-                    {
-                        var tree = new WatchTree
-                        {
-                            DataContext = new WatchViewModel(nodeViewModel.DynamoViewModel.BackgroundPreviewViewModel.AddLabelForPath)
-                        };
-                        tree.treeView1.ItemContainerGenerator.StatusChanged += WatchContainer_StatusChanged;
+                    DataContext = new WatchViewModel(nodeViewModel.DynamoViewModel.BackgroundPreviewViewModel.AddLabelForPath)
+                };
+                tree.treeView1.ItemContainerGenerator.StatusChanged += WatchContainer_StatusChanged;
 
-                        largeContentGrid.Children.Add(tree);
-                    }
+                largeContentGrid.Children.Add(tree);
+            }
 
-                    var watchTree = largeContentGrid.Children[0] as WatchTree;
-                    var rootDataContext = watchTree.DataContext as WatchViewModel;
+            var watchTree = largeContentGrid.Children[0] as WatchTree;
+            var rootDataContext = watchTree.DataContext as WatchViewModel;
 
+            cachedLargeContent = newViewModel;
 
-                    cachedLargeContent = newViewModel;
+            rootDataContext.IsOneRowContent = cachedLargeContent.Children.Count == 0;
+            rootDataContext.Children.Clear();
+            rootDataContext.Children.Add(cachedLargeContent);
 
-                    rootDataContext.IsOneRowContent = cachedLargeContent.Children.Count == 0;
-                    rootDataContext.Children.Clear();
-                    rootDataContext.Children.Add(cachedLargeContent);
+            watchTree.treeView1.SetBinding(ItemsControl.ItemsSourceProperty,
+                new Binding("Children")
+                {
+                    Mode = BindingMode.TwoWay,
+                    Source = rootDataContext,
+                });
 
-                    watchTree.treeView1.SetBinding(ItemsControl.ItemsSourceProperty,
-                        new Binding("Children")
-                        {
-                            Mode = BindingMode.TwoWay,
-                            Source = rootDataContext
-                        });
-                    if (refreshDisplay != null)
+            if (refreshDisplay != null)
+            {
+                refreshDisplay();
+            }
+        }
+
+        /// <summary>
+        /// Gets condensed string asynchronously, so that UI thread is not frozen.
+        /// </summary>        
+        private Task<string> GetCondensedStringAsync()
```

Comment:
I did not quite get it. As I have not seen any computation that is heavy here, why need this to run asynchronously?